metabulo/models.py

```diff
@@ -248,28 +253,34 @@ def make_object(self, data):
 
 class TableColumn(db.Model):
     __table_args__ = (
-        UniqueConstraint('column_header', 'csv_file_id'),
+        UniqueConstraint('id', 'csv_file_id'),
     )
-
+    id = db.Column(UUIDType(binary=False), default=uuid4)
     csv_file_id = db.Column(UUIDType(binary=False), db.ForeignKey('csv_file.id'), primary_key=True)
     column_index = db.Column(db.Integer, primary_key=True)
-    column_header = db.Column(db.String, nullable=False)
     column_type = db.Column(db.String, nullable=False)
 
+    @property
+    def column_header(self):
+        if self.csv_file.index_row is not None:
+            index = self.csv_file.index_row.row_index
+            return self.csv_file.table.iat[index, self.column_index]
+        return ''
```

Comment:
@jbeezley yeah, it was a doozy.  My approach was to remove `column_header` and `row_name` from the data model and make them "computed" properties.   Then, when you change a key, it's no different than changing metadata or masks, which is super nice.

The Catch (TM)  is that this relies total removal of reliance on pandas' indexes.    I currently still believe that breaking with the pandas data model is inevitable, so we might as well get it over with, because it's just going to make life harder later.

There _is_ a hybrid approach where we reorder the array with the indexes in row/col0 when it's read from disk.  To borrow my new favorite phrase from Zach, that gave me bad tummy feels.

Reply:
I still think we can avoid abandoning the pandas indexing without too many bad tummy feels, but I might just have to prove myself wrong when I attempt it.